Azure-Samples/modern-data-warehouse-dataops
# Update devcontainer with java and non-root user

# Type of PR
- Code changes

## Purpose
Fix issue #644 - Devcontainer failing with unsupported JDK
Fix issue #646 - Devcontainer changes file ownership in linux/wsl
Updates devcontainer to bring Critical and High CVE vulnerabilities reported to zero

## Does this introduce a breaking change? If yes, details on what can break
No

## Validation steps
- Clone branch
- Open VSCode folder in DevContainer
- In devcontainer terminal prompt, verify java version: `java -version`
- In devcontainer terminal prompt, run the command `touch test.file`
- In local WSL prompt, navigate to `/mnt/.../e2e_samples/parking_sensors` (replace "..." with location where you cloned the code)
- Validate you can edit the test.file without requiring root permission.

## Issues Closed or Referenced
- Closes #644, #646
